daemon/lxd.go

```diff
@@ -166,3 +169,92 @@ func (l *LXD) DeleteContainer(hostname string) (bool, error) {
 
 	return true, nil
 }
+
+func (l *LXD) CreateContainerFile(container pfmodel.Container, fullPath string) error {
+	var content string
+	mode := 644
+	contentType := "file"
+	writeMode := "overwrite"
+
+	// // prepare filescript
+	for _, bs := range container.Bootstrappers {
+		switch bs.Type {
+		case "chef-solo":
+			content = `
+cd /tmp && curl -LO https://www.chef.io/chef/install.sh && sudo bash ./install.sh -v 14.12.3 && rm install.sh
+cat > solo.rb << EOF
+root = File.absolute_path(File.dirname(__FILE__))
+cookbook_path root + "/cookbooks"
+EOF
+`
+			execChefSoloCmd := fmt.Sprintf("chef-solo -c ~/tmp/solo.rb -j %s %s", bs.Attributes, bs.CookbooksUrl)
+			content = content + "\n" + execChefSoloCmd
+			mode = 600
+		default:
+			content = content
+			mode = mode
+		}
+	}
+
+	bootstrapFile, err := util.WriteToFile(fullPath, content)
+	if err != nil {
+		return err
+	}
+
+	// Setup the various options for a container file upload
+	fileArgs := client.ContainerFileArgs{
+		Content:   bootstrapFile,
+		UID:       0,
+		GID:       0,
+		Mode:      mode,
+		Type:      contentType,
+		WriteMode: writeMode,
+	}
+
+	err = l.targetSrv.CreateContainerFile(container.Hostname, fullPath, fileArgs)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (l *LXD) ExecContainer(container pfmodel.Container, fullPath string) (bool, error) {
```

Comment:
please fix the naming, not showing intent

Reply:
done